In [17]:
satisfaction = read.table("/home/isaac/R Projects, STAC67/A2, Q5")

In [18]:
satisfaction

V1 V2 V3 V4 
1  48 50 51 2.3
2  57 36 46 2.3
3  66 40 48 2.2
4  70 41 44 1.8
5  89 28 43 1.8
6  36 49 54 2.9
7  46 42 50 2.2
8  54 45 48 2.4
9  26 52 62 2.9
10 77 29 50 2.1
11 89 29 48 2.4
12 67 43 53 2.4
13 47 38 55 2.2
14 51 34 51 2.3
15 57 53 54 2.2
16 66 36 49 2.0
17 79 33 56 2.5
18 88 29 46 1.9
19 60 33 49 2.1
20 49 55 51 2.4
21 77 29 52 2.3
22 52 44 58 2.9
23 60 43 50 2.3
24 86 23 41 1.8
25 43 47 53 2.5
26 34 55 54 2.5
27 63 25 49 2.0
28 72 32 46 2.6
29 57 32 52 2.4
30 55 42 51 2.7
31 59 33 42 2.0
32 83 36 49 1.8
33 76 31 47 2.0
34 47 40 48 2.2
35 36 53 57 2.8
36 80 34 49 2.2
37 82 29 48 2.5
38 64 30 51 2.4
39 37 47 60 2.4
40 42 47 50 2.6
41 66 43 53 2.3
42 83 22 51 2.0
43 37 44 51 2.6
44 68 45 51 2.2
45 59 37 53 2.1
46 92 28 46 1.8

In [19]:
anxiety = satisfaction$V4

In [20]:
severity = satisfaction$V3

In [21]:
age = satisfaction$V2

In [22]:
satisfaction_var = satisfaction$V1

In [29]:
fit1 = lm(satisfaction_var~age+anxiety+severity, data = satisfaction)

In [30]:
anova(fit1)

Df Sum Sq     Mean Sq    F value    Pr(>F)      
age        1 8275.38885 8275.38885 81.8026275 2.059138e-11
anxiety    1  763.41576  763.41576  7.5464025 8.819116e-03
severity   1   81.65905   81.65905  0.8072038 3.740702e-01
Residuals 42 4248.84068  101.16287         NA           NA

In [31]:
summary(fit1)


Call:
lm(formula = satisfaction_var ~ age + anxiety + severity, data = satisfaction)

Residuals:
     Min       1Q   Median       3Q      Max 
-18.3524  -6.4230   0.5196   8.3715  17.1601 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept) 158.4913    18.1259   8.744 5.26e-11 ***
age          -1.1416     0.2148  -5.315 3.81e-06 ***
anxiety     -13.4702     7.0997  -1.897   0.0647 .  
severity     -0.4420     0.4920  -0.898   0.3741    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 10.06 on 42 degrees of freedom
Multiple R-squared:  0.6822,	Adjusted R-squared:  0.6595 
F-statistic: 30.05 on 3 and 42 DF,  p-value: 1.542e-10


# Conclusions:

H_0 : beta_1 = beta_2 = beta_3 = 0

H_a : not all beta_k = 0 s.t. k = 1,2,3


From summary(fit1), we have F-statistic = 30.05. Then, F(0.90, 3, 42) = 2.2191. If F-statistic <= 2.2191, conclude H_0; o/w, conclude H_a. 30.05 > 2.2191, reject the null hypothesis. Therefore, some of the beta_k = 0 (k = 1, 2, 3). Additionally, p-value = 1.542e-10 supports our argument of rejecting the null hypothesis.


In [33]:
coefficients(fit1)

(Intercept)         age     anxiety    severity 
158.4912517  -1.1416118 -13.4701632  -0.4420043

In [34]:
confint(fit, level = 0.90)

5 %        95 %       
(Intercept) 128.004370 188.9781330
V2           -1.502893  -0.7803305
V3           -1.269467   0.3854587
V4          -25.411454  -1.5288719

- b1_st_e = 0.2148
- b2_st_e = 0.4920
- b3_st_e = 7.0097


- t(1 - 0.1/(2x3), 42) = 2.1995

- beta_1: -1.1416 (+-) 2.1995 x 0.2148 => -1.5968804 <= beta_1 <= -0.6863432
- beta_2: -0.4420043 (+-) 2.1995 x 0.4920 => -1.524183 <= beta_2 <= 0.6401497
- beta_3: -13.4701632 (+-) 2.1995 x 7.0997 => -29.08595335 <= beta_3 <= 2.14562695

- R_Squared (aka, coefficient of multiple determination) = 0.6822
=> 68.22% is the proportion of variability in satisfaction explained by the regression model with patient's age (V2), severity of illness (V3), and anxiety level (V4).